lib/UserDiscoveryModules/RemoteUserMapping.php

```diff
@@ -30,22 +33,31 @@ public function __construct(
 		private LoggerInterface $logger,
 	) {
 		$this->discoveryEndpoint = $config->getSystemValueString('gss.discovery.remote.endpoint', '');
+		$this->discoverySecretKey = $config->getSystemValueString('gss.discovery.remote.secret', '');
 
 		$this->logger->debug('Init RemoteUserMapping');
 		$this->logger->debug('host: ' . $this->discoveryEndpoint);
 	}
 
 	public function getLocation(array $data): string {
 		$client = $this->clientService->newClient();
-		$location = '';
+		if ($this->discoverySecretKey !== '') {
+			$data['gsSecretKey'] = $this->discoverySecretKey;
+		}
+
 		try {
 			$result = $client->post($this->discoveryEndpoint, ['body' => $data]);
-			$location = $result->getBody();
 		} catch (\Exception $e) {
 			$this->logger->warning('cannot access remote discovery endpoint ' . $this->discoveryEndpoint, ['exception' => $e]);
 		}
 
-		$this->logger->debug('extracted location from remote discovery: ' . $location);
-		return $location;
+		try {
+			$location = json_decode($result->getBody(), true, flags: JSON_THROW_ON_ERROR);
+		} catch (\JsonException) {
+			$this->logger->warning('cannot parse remote discovery endpoint result', ['result' => $result->getBody()]);
+		}
+
+		$this->logger->debug('extracted location from remote discovery: ' . json_encode($location));
+		return $location['location'] ?? '';
```

Comment:
Is there a fallback later if no location is found in the response payload (if '' is returned)? If not, we should probably throw an exception here.

Reply:
yes, empty location are detected and an HintException is thrown